src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -447,12 +447,14 @@ void RTPSParticipantImpl::enable()
 
 void RTPSParticipantImpl::disable()
 {
-    // Ensure that other participants will not accidentally discover this one
-    if (mp_builtinProtocols && mp_builtinProtocols->mp_PDP)
+    if (nullptr == mp_builtinProtocols)
     {
-        mp_builtinProtocols->stopRTPSParticipantAnnouncement();
+        return;
     }
 
+    // Ensure that other participants will not accidentally discover this one
+    stopRTPSParticipantAnnouncement();
+
```

Comment:
Removing the `mp_builtinProtocols->mp_PDP` check will log an error when trying to disable a participant with `DiscoveryProtocol::None`. Maybe we should only call `stopRTPSParticipantAnnouncement` with a valid `mp_builtinProtocols->mp_PDP` here (as it was done before).

Reply:
Absolutely right. I introduced the fixed in ec657ef